src/main/kotlin/org/rust/lang/utils/evaluation/CfgEvaluator.kt

```diff
@@ -87,24 +86,29 @@ class CfgEvaluator(
         is CfgPredicate.Not -> !evaluatePredicate(predicate.single)
         is CfgPredicate.NameOption -> evaluateName(predicate.name)
         is CfgPredicate.NameValueOption -> evaluateNameValue(predicate.name, predicate.value)
-        is CfgPredicate.Feature -> evaluateFeature(predicate.name)
         is CfgPredicate.Error -> Unknown
     }
 
     private fun evaluateName(name: String): ThreeValuedLogic = when {
-        name in packageOptions.nameOptions -> True
-        // TODO: convert whitelist to blacklist and merge options with packageOption
-        name in SUPPORTED_NAME_OPTIONS -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
-        (name == "test" || name == "bootstrap" || name == "doc") && origin == PackageOrigin.STDLIB -> False
-        name == CfgOptions.TEST && isUnitTestMode -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
-        else -> Unknown
+        // https://doc.rust-lang.org/nightly/unstable-book/language-features/cfg-panic.html
+        name == "cfg_panic" -> Unknown
+
+        name == "test" && origin != PackageOrigin.STDLIB -> Unknown
+        strict -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
+        else -> when {
+            name in SUPPORTED_NAME_OPTIONS -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
+            (name == "test" || name == "bootstrap" || name == "doc") && origin == PackageOrigin.STDLIB -> False
```

Comment:
Note that `test` and `doc` options are in `SUPPORTED_NAME_OPTIONS` so they are processed by the first match arm. Consider something like
```Kotlin
when {
    name in STDLIB_FALSE_NAME_OPTIONS && origin == PackageOrigin.STDLIB -> False
    name in SUPPORTED_NAME_OPTIONS -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
    else -> Unknown
}
```